src/Migrations/M033MigrateCampaignCoversToP4ColumnsBlock.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+namespace P4\MasterTheme\Migrations;
+
+use P4\MasterTheme\MigrationRecord;
+use P4\MasterTheme\MigrationScript;
+
+/**
+ * Migrate Campaign covers block to Planet4 columns block.
+ */
+class M033MigrateCampaignCoversToP4ColumnsBlock extends MigrationScript
+{
+    /**
+     * Extract campaign covers block from page/posts and transform it into Planet4 columns block.
+     *
+     * @param MigrationRecord $record Information on the execution, can be used to add logs.
+     * phpcs:disable SlevomatCodingStandard.Functions.UnusedParameter -- interface implementation
+     */
+    public static function execute(MigrationRecord $record): void
+    {
+        $check_is_valid_block = function ($block) {
+            return self::check_is_valid_block($block);
+        };
+
+        $transform_block = function ($block) {
+            return self::transform_block($block);
+        };
+
+        Utils\Functions::execute_block_migration(
+            Utils\Constants::BLOCK_COVERS,
+            $check_is_valid_block,
+            $transform_block,
+        );
+    }
+    // phpcs:enable SlevomatCodingStandard.Functions.UnusedParameter
+
+    /**
+     * Check whether a block is a Campaign Covers block.
+     *
+     * @param array $block - A block data array.
+     */
+    private static function check_is_valid_block(array $block): bool
+    {
+        // Check if the block is valid.
+        if (!is_array($block)) {
+            return false;
+        }
+
+        // Check if the block has a 'blockName' key.
+        if (!isset($block['blockName'])) {
+            return false;
+        }
+
+        // Check if the block is a Cover block. If not, abort.
+        if ($block['blockName'] !== Utils\Constants::BLOCK_COVERS) {
+            return false;
+        }
+
+        // For older cover blocks where the cover type is empty,
+        // the default content cover type is applied to those blocks.
+        if (!isset($block['attrs']['cover_type'])) {
+            return false;
+        }
+
+        // For old cover type(earlier it was numeric).
+        if (is_numeric($block['attrs']['cover_type'])) {
+            // phpcs:ignore Generic.Files.LineLength.MaxExceeded
+            $block['attrs']['cover_type'] = Utils\Constants::OLD_COVER_TYPES[ $block['attrs']['cover_type'] ];
```

Comment:
Maybe we could use this migration to completely get rid of the old numeric values in the rest of our code?

Reply:
I am not sure of understanding this, can you please explain it further?

Reply:
Maybe in this migration we can also migrate Content and Take Action covers blocks that still use the old numeric values, so that they now use the new string ones. And then we could get rid of [this](https://github.com/greenpeace/planet4-master-theme/blob/f8eca142768fb28ff32ec7d6f9fb98e4026e2769/src/Blocks/Covers.php#L31-L38) for example.

Reply:
I'd rather do it on another PR.
I don't want to make this one more confusing by adding 2 types of migrations: one for the Campaign styles and another for the Content and Take Action styles.
What do you think?

Reply:
Yeah that's fair, we can investigate this later. Especially because we will also be migrating other types of Covers.